encoding/thrift/inbound_nowire.go

```diff
@@ -0,0 +1,207 @@
+// Copyright (c) 2021 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package thrift
+
+import (
+	"context"
+	"io"
+
+	"go.uber.org/thriftrw/protocol/binary"
+	"go.uber.org/thriftrw/protocol/stream"
+	"go.uber.org/thriftrw/wire"
+	encodingapi "go.uber.org/yarpc/api/encoding"
+	"go.uber.org/yarpc/api/transport"
+	"go.uber.org/yarpc/pkg/errors"
+)
+
+var _emptyResponse = NoWireResponse{}
+
+// thriftNoWireHandler is similar to thriftUnaryHandler and thriftOnewayHandler
+// except that thriftNoWireHandler implements both transport.UnaryHandler and
+// transport.OnewayHandler through a single type, utilizing the "nowire"
+// (streaming in the ThriftRW context, bypassing the intermediate "Wire"
+// representation) implementation.
+//
+// The UnaryHandler and OnewayHandler implementations then call into type that
+// implements NoWireHandler, which understands how to unpack and invoke the
+// request, given the ThriftRW primitives for reading the raw representation.
+type thriftNoWireHandler struct {
+	NoWireHandler NoWireHandler
```

Comment:
We're inside a struct called thriftNoWireHandler, so we don't need to name the field NoWireHandler. It can just be "Handler" or similar.